core/src/main/scala/com/raphtory/api/input/sources/CSVEdgeListSource.scala

```diff
@@ -0,0 +1,98 @@
+package com.raphtory.api.input.sources
+
+import com.raphtory.api.input.Graph.assignID
+import com.raphtory.api.input.{Graph, GraphBuilder, ImmutableProperty, Properties, Source, SourceInstance, Spout, SpoutInstance}
+import com.raphtory.internals.graph.GraphBuilderInstance
+import com.raphtory.spouts.{FileSpout, ResourceSpout}
+import com.raphtory.internals.time.DateTimeParser.{defaultParse => parseDateTime}
+
+import scala.language.implicitConversions
+
+class CSVEdgeListSource(override val spout: Spout[String], timeIndex: Int = 2, sourceIndex: Int = 0, targetIndex: Int = 1) extends Source {
+  override type MessageType = String
+
+  private var dateTimeFormat: Boolean = _
+  private var epochFormat: Boolean = _
+
+  private var longFormat: Boolean = _
+  private var stringFormat: Boolean = _
+
+
+  def buildCSVEdgeListGraph(graph: Graph, rawTime: String, source: String, target: String) = {
+    val timestamp = {
+      if (dateTimeFormat) {
+        parseDateTime(rawTime)
+      } else if (epochFormat) {
+        rawTime.toLong
+      } else {
+        throw new RuntimeException("Check timestamp is in Epoch or DateTime Format")
+      }
+    }
+
+    val src = {
+      if (longFormat) {
+        source.toLong
+      } else if (stringFormat) {
+        assignID(source)
+      } else {
+        throw new RuntimeException("Check source is in Long or String format")
```

Comment:
Similar to above and for the one below can this say the src/dst id which is broken and the tuple it is from? Makes it much easier for a user to find it.